test/e2e/performanceprofile/functests/8_performance_workloadhints/workloadhints.go

```diff
@@ -289,7 +289,7 @@ var _ = Describe("[rfe_id:49062][workloadHints] Telco friendly workload specific
 					"vm.stat_interval":              "10",
 				}
 				kernelParameters := []string{noHzParam, "tsc=reliable", "nosoftlockup", "nmi_watchdog=0", "mce=off", "skew_tick=1",
-					"processor.max_cstate=1", "intel_idle.max_cstate=0", "intel_pstate=disable", "idle=poll"}
+					"processor.max_cstate=1", "intel_idle.max_cstate=0", "intel_pstate=${automatic_pstate}", "idle=poll"}
```

Comment:
@MarSik Do we know where or in what kind of machine this `e2e-gcp-pao-workloadhints` test runs? If tuned hasn't released a new version, the `intel_recommended_pstate` function will not be evaluated. In addition, without knowing underlying hw, we also don't know whether `${automatic_pstate}` will be resolved to active or disable.

Reply:
@MarSik So i found out that in this prow job, functional tests run, and from the log i can see that the kernel parameters set in the node are
`<string>: BOOT_IMAGE=(hd0,gpt3)/ostree/rhcos-653f1642375587f2cc93c3a0d201773ed3fd73c0567f2cc8fea333e1f2d236c4/vmlinuz-5.14.0-427.el9.x86_64+rt ostree=/ostree/boot.0/rhcos/653f1642375587f2cc93c3a0d201773ed3fd73c0567f2cc8fea333e1f2d236c4/0 ignition.platform.id=gcp ......intel_pstate= {f:intel_recommended_pstate}`

I think what happened here is 
- tuned doesn't have this function yet, so it can't resolve. i.e. **intel_pstate= {f:intel_recommended_pstate}**
- when the function will be available, do we know whether this test will run in pre-IceLake or IceLake+ servers?

Also, i think my proposed changes in the workloadhints.go is wrong, because we will either have `disable|active` in those cases where i put `intel_pstate=${automatic_pstate}`

I can change `intel_pstate=${automatic_pstate}` to `intel_pstate= {f:intel_recommended_pstate}` in the kernelParams to pass the test, but this test will fail once the function becomes available in the tuned side.

or may be, i'm mistaken, these tests probably never rely on tuned at all?

How do we proceed?

Reply:
Hmm, this is an interesting issue. We actually do not guarantee a specific platform. @mrniranjan Any idea how to make the test hardware aware?

Reply:
Currently to execute tests only hardware we verify it the worker nodes  have  2 numa nodes and number of cpus is more than 32.

Reply:
The kernel argument test will have to be improved to read the information about what CPU was used.